Sources/MongoKitten/AggregateBuilder.swift

```diff
@@ -0,0 +1,174 @@
+import BSON
+
+public final class SubAggregateBuilder: AggregateBuilder {
+    public var pipeline: [Document] = []
+}
+
+public protocol AggregateBuilder: class {
+    var pipeline: [Document] { get set }
+    
+    /// Limits the number of documents passed to the next stage in the pipeline.
+    ///
+    /// - parameter limit: A positive integer that specifies the maximum number of documents to pass along.
+    @discardableResult func limit(_ limit: Int) -> Self
```

Comment:
I think we can remove these definitions from the aggregatebuilder spec and just implement it in an extension only. The requirement doesn't add any benefits here.

Reply:
Don't forget to move the docs to the extension!